tree-sitter-darklang/grammar.js

```diff
@@ -290,6 +291,26 @@ module.exports = grammar({
     // Floats
     float_literal: $ => /[+-]?[0-9]+\.[0-9]+/,
 
+    //
+    // List
+    // TODO: allow multi-line lists where a newline is 'interpreted' as a list delimiter (i.e. no ; needed)
+    // TODO: support [1; 2;] (lists ending with a ; -- as part of being a 'tolerant' parser)
+    list_literal: $ =>
+      seq(
+        field("symbol_open_bracket", alias("[", $.symbol)),
+        field("content", optional($.list_content)),
+        field("symbol_close_bracket", alias("]", $.symbol)),
+      ),
+
+    list_content: $ =>
```

Comment:
Couldn't use the following rule, because this will pass [1 2]
```
    list_content: $ =>
      repeat1(
        seq(
          $.expression,
          optional(field("list_separator", alias(";", $.symbol))),
        ),
      ),
 ```

Reply:
Cool - I like how this turned out, anyway -- better represents the situation we're trying to capture.